src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php

```diff
@@ -0,0 +1,89 @@
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Controller\Admin\Sell\Catalog\Product;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Query\GetEditableCombinationsList;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\CombinationListForEditing;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+class CombinationController extends FrameworkBundleAdminController
+{
+    /**
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))")
+     *
+     * @param int $productId
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function getListAction(int $productId, Request $request): JsonResponse
+    {
+        $limit = (int) $request->query->get('limit');
+        $page = (int) $request->query->get('page');
+
+        $combinationsList = $this->getQueryBus()->handle(new GetEditableCombinationsList(
+            $productId,
+            $this->getContextLangId(),
+            $limit ?? null,
+            $page ?? null
+        ));
+
+        return $this->json($this->formatResponse($combinationsList));
+    }
+
+    /**
+     * @param CombinationListForEditing $combinationListForEditing
+     *
+     * @return array<string, array<int, array<string,bool|int|string>>|int>
+     */
+    private function formatResponse(CombinationListForEditing $combinationListForEditing): array
+    {
+        $data = [
+            'combinations' => [],
+            'total' => $combinationListForEditing->getTotalCombinationsCount(),
+        ];
+        foreach ($combinationListForEditing->getCombinations() as $combination) {
+            $data['combinations'][] = [
+                'id' => $combination->getCombinationId(),
+                'isSelected' => false,
+                'name' => $combination->getCombinationName(),
+                //@todo: don't forget image path when implemented in the query
+                'impactOnPrice' => (string) $combination->getImpactOnPrice(),
+                //@todo: calculate final price. Need a service to be used in formData provider and here
```

Comment:
This can be discussed later, but I'm not sure the `finalpriceTe` needs to be returned by the API, if I get it right it's the sum of product price 'tax excluded) and the impact on price And it's only displayed for the user's conveniency

So I think it could be handled by javascript in the front, it would have two advantages:
- avoid making this API controller more complicated
- if the price is modified in the front then the combination final price can be automatically updated

TLDR; So instead of getting this final price from the API, the combination component could get the product price from the product form and adapt its display dynamically

Reply:
Anyway it's to be done in another PR